### PR TITLE
feat: add tokenId to subscriber message response

### PR DESCRIPTION
**Pull request by pratik151192**
For momentohq/dev-eco-issue-tracker/issues/511

### CODE REVIEW

File: packages/core/src/messages/responses/topic-item.ts [hunk 1]
```diff
@@ -36,8 +40,22 @@ export class TopicItem {
     return this.value() as Uint8Array;
   }
 
+  /**
+   * Optionally returns the publisher ID from the steam if it exists.
+   * @returns string | undefined
+   */
+  public publisherID(): string | undefined {
```
Comment (anitarua) [inline packages/core/src/messages/responses/topic-item.ts:47]: tiny nit: `publisherID` --> `publisherId` ?

**Review by anitarua — APPROVED** [review]
Overall looks good to me!

File: packages/core/src/messages/responses/topic-item.ts [hunk 2]
```diff
@@ -36,8 +40,22 @@ export class TopicItem {
...
+    return this._publisherID;
+  }
+
   public toString(): string {
-    const display = truncateString(this.value().toString());
-    return `${this.constructor.name}: ${display}`;
+    const displayValue = truncateString(this.value().toString());
+    let displayString = `${this.constructor.name}: ${displayValue}`;
+
+    if (this._publisherID) {
```
Comment (pratik151192) [inline packages/core/src/messages/responses/topic-item.ts:55]: will add the `undefined` check explicitly

File: packages/core/src/messages/responses/topic-item.ts (hunk 1 above)
Comment (pratik151192) [reply packages/core/src/messages/responses/topic-item.ts:47]: yes good catch will do

**Review by anitarua — APPROVED** [review]

**Review by cprice404 — CHANGES_REQUESTED** [review]

**Review by cprice404 — APPROVED** [review]

**Pull request merged**